extensions/storage/README.md

```diff
@@ -0,0 +1,78 @@
+# Object Storage Extension Specification
+
+- **Title: Object Storage**
+- **Identifier: storage**
+- **Field Name Prefix: -**
+- **Scope: Item, Assets**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+- **Owner**: @davidraleigh
+
+This document explains the fields of the Object Storage Extension to a STAC Item and an Asset. This does not cover NFS solutions provided by PaaS cloud companies. Allows user to specify details related to cloud storage access and costs for assets and their STAC item.
+
+- [Example (Landsat 8)](examples/example-landsat8.json)
+- [JSON Schema](json-schema/schema.json)
+
+## Asset fields
+
+| Field Name  | Type   | Description |
+| ----------- | ------ | ----------- |
+| storage:cloud_platform        | string    | The [cloud provider](#providers) where data is stored |
+| storage:manager               | string    | The entity in charge of managing the data. |
+| storage:regions               | [string]  | The region(s) where the data is stored. Relevant to speed of access and inter region egress costs (as defined by PaaS provider) |
+| storage:bucket                | string    | The bucket for the asset, used along with object path |
+| storage:object_path           | string    | The object_path for the asset, used along with bucket |
+| storage:requester_pays        | bool      | Is the data requester pays or is it data manager/cloud provider pays |
+| storage:tier                  | string    | The title for the tier type (as defined by PaaS provider) |
+| storage:tier_duration         | integer   | Minimum storage duration required before additional fees |
+| storage:date_stored           | string    | Date and time the corresponding asset placed into the current storage tier (relevant for tier_duration > 0) |
+| storage:first_byte_latency    | string    | time unit (milliseconds, minutes or hours) for accessing first byte of data |
+
```

Comment:
```suggestion
**storage:cloud_platform** value representing the name of the object store. Valid values for common cloud storage providers are `aws_s3`, `alibaba_oss`, `microsoft_azure_storage`, `google_cs`, `ibm_os`, and `oracle_os`
```

Reply:
How can we make that in the future we don't need to change the spec if a new cloud service appears? The ones here sound a bit US centric.

Reply:
Maybe better have a table below with common services and their identifier, but make that list non-exclusive?

Reply:
Yes, agree -- I phrased that wrong in text. I meant that for these, providers, use these values -- but there can and will be more

Reply:
Does it need to be `aws_s3` or is `s3` sufficient, where we can rely on established protocol names already used in URLs? 
e.g., s3://, gs://   

`s3` for AWS and `gs` for Google seem to be widely used, there just doesn't seem to be an accepted upon standard for Azure.

Reply:
I think I'd also like to use the protocol names, although several providers implement s3. Do they all use the same protocol then?

Reply:
I think implementing s3 just means you conform to the AWS S3 REST API that clients (such as the cli) use anyway. 

I'd prefer separating the scheme from the actual service being used, and to follow the general pattern of `{provider}_{service_name}` -- otherwise, we'd have to assume `aws` meant `s3` (what if they later introduce `s4`?), whereas the other more generic names are just like `cloud_storage` have to have the provider. So, consistency.

Also, the Asset#href might be an ARN (`arn:aws:s3:::bucket_name/key_name`) and then you're back to parsing the URI to find the service.